### ISSUE TITLE
verify(fix=true) returns wrong violations after multi-pass fix

### ISSUE THREAD

**Issue by saphryn** [markuplint/markuplint#3890]
## Bug Description

`MLEngine.verify({ fix: true })` returns a `violations` array that reflects the **first** fix pass only, but `fixedCode` may be the result of **pass N** (N > 1). Violations introduced during later fix passes are silently dropped. This means callers who check the `violations` array to determine remaining problems after fixing get a false picture.

## Root Cause

In `packages/@markuplint/ml-core/src/ml-core.ts`, the `#multiPassFix` method iteratively applies fixes, re-parses, and re-runs rules until convergence or MAX_FIX_PASSES (10). However, the `violations` array returned by `verify()` is collected **before** the multi-pass loop starts:

```typescript
// verify() collects violations from FIRST pass only
const ruleViolations = await this.#runAllRules(fix);
violations.push(...ruleViolations);

// Then multi-pass fix runs (modifies state, restores in finally)
const fixResult = await this.#multiPassFix(violations);
fixedCode = fixResult.code;
```

The JSDoc explicitly acknowledges this:
> *"Callers needing an accurate violation list for the fixed code should re-verify the output."*

But no mechanism exists to do this automatically, and no warning is issued.

## Impact

- **CI false pass**: A pipeline that checks `result.violations` after `verify({fix:true})` will incorrectly conclude no violations remain, even when the fixed code has introduced new ones in later passes.
- **Developer confusion**: The violations array and the fixed code are inconsistent by design.
- **Silent data loss**: Violations from later passes are simply discarded with no indicator.

## Reproduction

1. Create markup where two fixes interact across passes — e.g., Fix A in pass 1 changes structure, causing a new Violation B in pass 2.
2. Call `verify({ fix: true })`
3. Observe: `violations` does not contain Violation B, but `fixedCode` reflects the changed structure.

## Proposed Fix

**Option A (recommended)**: Return `remainingViolations` from the final pass alongside `violations` (rename to `initialViolations`). `verify()` would then return:
```typescript
type VerifyResult = {
  initialViolations: readonly Violation[];  // first pass
  remainingViolations: readonly Violation[]; // final pass  
  fixedCode: string | undefined;
  fixSummary?: FixSummary;
}
```

**Option B**: Have `verify({ fix: true })` internally re-run `verify(false)` on the fixed code and merge the results, with a flag to prevent infinite loops.

## Additional Context

This bug is tracked in the context of the Rust rewrite (v6). The Fix phase (Phase 3-3~3-6) has not yet started per https://github.com/markuplint/markuplint/issues/3178. Fixing this in the TS core would benefit users before v6 lands.

Affected file: `packages/@markuplint/ml-core/src/ml-core.ts`

**Comment by YusukeHirao** [comment]
Thanks for the detailed report. Triaged and confirmed against the source: the `violations` array is collected before `#multiPassFix` runs and is never updated afterwards, so the CLI exit code and `--suppress` generation operate on first-pass violations even when later passes changed the outcome.

**Decision: fix in v5 RC** (non-breaking variant of Option A/B):

- Add an optional `finalPassViolations` field to `FixSummary`. On the multi-pass path the final-pass violations are already computed inside `#multiPassFix` (`#runAllRules(true)`) and currently discarded — they will simply be plumbed out. On single-pass exits, one extra verification run will be performed when at least one fix was applied.
- The CLI will use `finalPassViolations ?? violations` for reporting, exit code, and `--suppress`.
- The existing `violations` field keeps its current semantics (first pass), so this is non-breaking. The API split proposed as Option A is deferred to the v6 fix-phase design (#3178).

A PR will follow, combined with the cycle-detection fix for #3891 since both touch `#multiPassFix`.
